src/collection/backend/lmdb.cc

```diff
@@ -498,22 +499,28 @@ void LMDB::resolveMultiMatches(const std::string& var,
         while ((rc = mdb_cursor_get(cursor, &key, &data, MDB_NEXT)) == 0) {
             l->insert(l->begin(), new VariableValue(
                 &m_name,
-                new std::string(reinterpret_cast<char *>(key.mv_data),
+                std::string(reinterpret_cast<char *>(key.mv_data),
                 key.mv_size),
-                new std::string(reinterpret_cast<char *>(data.mv_data),
+                std::string(reinterpret_cast<char *>(data.mv_data),
                 data.mv_size)));
         }
     } else {
-        while ((rc = mdb_cursor_get(cursor, &key, &data, MDB_NEXT)) == 0) {
-            char *a = reinterpret_cast<char *>(key.mv_data);
-            if (strncmp(var.c_str(), a, keySize) == 0) {
-                l->insert(l->begin(), new VariableValue(
-                    &m_name,
-                    new std::string(reinterpret_cast<char *>(key.mv_data),
-                    key.mv_size),
-                    new std::string(reinterpret_cast<char *>(data.mv_data),
-                    data.mv_size)));
-            }
+        string2val(var, &key);
+        rc = mdb_cursor_get(cursor, &key, &data, MDB_SET_RANGE);
+        if (rc == 0) {
+            do {
+                char *a = reinterpret_cast<char *>(key.mv_data);
+                if (strncmp(var.c_str(), a, keySize) == 0) {
+                    l->insert(l->begin(), new VariableValue(
+                        &m_name,
+                        std::string(reinterpret_cast<char *>(key.mv_data),
+                        key.mv_size),
+                        std::string(reinterpret_cast<char *>(data.mv_data),
+                        data.mv_size)));
+                } else {
+                    break;
+                }
+            } while ((rc = mdb_cursor_get(cursor, &key, &data, MDB_NEXT)) == 0);
```

Comment:
replace `MDB_NEXT` to `MDB_NEXT_DUB` to iterate only over current key

Reply:
**MDB_NEXT_DUP** - 
http://www.lmdb.tech/doc/group__mdb.html#gga1206b2af8b95e7f6b0ef6b28708c9127a37d22eecd320d77fbba9f905483f99e5